apps/site/assets/ts/schedule/components/LineDiagram.tsx

```diff
@@ -24,72 +42,82 @@ const maybeAlert = (alerts: Alert[]): ReactElement<HTMLElement> | null => {
+  connection.name.startsWith("SL") ? "u-bg--silver-line" : "u-bg--bus";
 
 const LineDiagram = ({
   lineDiagram
-}: Props): ReactElement<HTMLElement> | null => (
-  <>
-    {lineDiagram.map(
-      ({
-        route_stop: routeStop,
-        stop_data: stopData,
-        alerts: stopAlerts
-      }: LineDiagramStop) => (
-        <div key={routeStop.id} className="m-schedule-line-diagram__stop">
-          <div className="m-schedule-line-diagram__card-left">
-            <div className="m-schedule-line-diagram__stop-name">
-              {maybeAlert(stopAlerts)}
-              {routeStop.name}
-            </div>
-            <div>
-              {maybeTerminus(stopData) && maybeTerminus(stopData)!.branch}
-            </div>
-            <div className="m-schedule-line-diagram__connections">
-              {routeStop.connections.map((route: Route) =>
-                route.type === 3 && !route.name.startsWith("SL") ? (
-                  <span
-                    key={route.id}
-                    className="c-icon__bus-pill m-schedule-line-diagram__connection u-bg--bus"
-                  >
-                    {route.name}
-                  </span>
-                ) : (
-                  <span key={route.id}>{modeIcon(route.id)}</span>
-                )
-              )}
-            </div>
-          </div>
-          <div>
-            <div className="m-schedule-line-diagram__features">
-              {routeStop.stop_features.includes("parking_lot")
-                ? parkingIcon("c-svg__icon-parking-default")
-                : null}
-              {routeStop.stop_features.includes("access")
-                ? accessibleIcon("c-svg__icon-acessible-default")
-                : null}
-              {routeStop.zone && (
-                <span className="c-icon__cr-zone">{`Zone ${
-                  routeStop.zone
-                }`}</span>
-              )}
+}: Props): ReactElement<HTMLElement> | null => {
+  const routeType = lineDiagram[0].route_stop.route!.type;
+
+  return (
+    <>
+      <h3>
+        {routeType === 0 || routeType === 1 || routeType === 2
+          ? "Stations"
+          : "Stops"}
+      </h3>
+      {lineDiagram.map(
+        ({ route_stop: routeStop, alerts: stopAlerts }: LineDiagramStop) => (
+          <div key={routeStop.id} className="m-schedule-line-diagram__stop">
+            <div className="m-schedule-line-diagram__card-left">
+              <div className="m-schedule-line-diagram__stop-name">
+                {maybeAlert(stopAlerts)}
+                <a href={`/stops/${routeStop.id}`}>
+                  <h4>{routeStop.name}</h4>
+                </a>
+              </div>
+              <div className="m-schedule-line-diagram__connections">
+                {filteredConnections(routeStop.connections).map(
+                  (route: Route) =>
+                    route.type === 3 ? (
+                      <a href={`/schedules/${route.id}/line`}>
```

Comment:
The parent `<a>` tag could be lifted up as a shared parent. Seems the props on it are the same in either condition.